src/main/java/com/vmware/vipclient/i18n/messages/service/CacheService.java

```diff
@@ -36,29 +37,28 @@ public CacheService(MessagesDTO dto) {
         if (c == null) {
             return null;
         } else {
-            return c.get(cacheKey);
+            return (MessageCacheItem) c.get(cacheKey);
         }
     }
 
-    public void addCacheOfComponent(Map<String, String> dataMap) {
+    public void addCacheOfComponent(MessageCacheItem itemToCache) {
         String cacheKey = dto.getCompositStrAsCacheKey();
         Cache c = VIPCfg.getInstance().getCacheManager().getCache(VIPCfg.CACHE_L3);
         if (c != null) {
-            c.put(cacheKey, dataMap);
+            c.put(cacheKey, itemToCache);
         }
     }
 
-    public void updateCacheOfComponent(Map<String, String> dataMap) {
+    public void updateCacheOfComponent(MessageCacheItem itemToCache) {
         String cacheKey = dto.getCompositStrAsCacheKey();
         Cache c = VIPCfg.getInstance().getCacheManager().getCache(VIPCfg.CACHE_L3);
         if (c != null) {
-            Map<String, String> oldmap = c.get(cacheKey);
-            if (oldmap == null) {
-                c.put(cacheKey, dataMap);
-            } else {
-                oldmap.putAll(dataMap);
-                c.put(cacheKey, oldmap);
+            MessageCacheItem cacheItem = (MessageCacheItem) c.get(cacheKey);
+            if (cacheItem == null) {
+            	cacheItem = new MessageCacheItem();
+            	c.put(cacheKey, cacheItem);
             }
+            cacheItem.addCacheItem(itemToCache);
```

Comment:
How about changing like this? It seems better.
            if (cacheItem == null) {
            	c.put(cacheKey, itemToCache);
            }
            else {
                cacheItem.addCacheItem(itemToCache);
            }